clang/test/SemaSYCL/virtual-function.cpp

```diff
@@ -0,0 +1,29 @@
+// RUN: %clang_cc1 -fsycl-is-device -fsycl-allow-virtual-functions -internal-isystem %S/Inputs -verify -fsyntax-only %s
```

Comment:
Without the flag the diagnostic called will be  "SYCL kernel cannot call a virtual function" which is unrelated to this PR but I can add the test to this file because it is good to have

Reply:
I removed the test I added in previous commit without the flag because I don't think it has anything to do with this PR. If you prefer it remains, I will add back in. Or I can do it in a followup PR.